code/modules/mob/living/carbon/human/species/plasmaman.dm

```diff
@@ -235,10 +236,24 @@
 			if(heat_level_3 to INFINITY)
 				H.apply_damage(HEAT_GAS_DAMAGE_LEVEL_3, BURN, "head", used_weapon = "Excessive Heat")
 				H.fire_alert = max(H.fire_alert, 2)
+	return 1
 
-	if(!istype(H.wear_suit, /obj/item/clothing/suit/space/eva/plasmaman) || !istype(H.head, /obj/item/clothing/head/helmet/space/eva/plasmaman))
-		to_chat(H, "<span class='warning'>Your body reacts with the atmosphere and bursts into flame!</span>")
-		H.adjust_fire_stacks(0.5)
-		H.IgniteMob()
+/datum/species/plasmaman/handle_life(var/mob/living/carbon/human/H)
```

Comment:
If you go with TG's system, then you can have everything handled purely by the plasmamen, here with something like this:

```
/datum/species/plasmaman/handle_life(var/mob/living/carbon/human/H)
	var/datum/gas_mixture/environment = H.loc.return_air()
	if(!istype(H.wear_suit, /obj/item/clothing/suit/space/eva/plasmaman) || !istype(H.head, /obj/item/clothing/head/helmet/space/eva/plasmaman))
		if(environment && environment.total_moles())
			if(environment.oxygen && environment.oxygen >= OXYCONCEN_PLASMEN_IGNITION)	
				H.adjust_fire_stacks(0.5)
				if(!H.on_fire && H.fire_stacks > 0)
					H.visible_message("<span class='danger'>[H]'s body reacts with the atmosphere and bursts into flames!</span>","<span class='userdanger'>Your body reacts with the atmosphere and bursts into flame!</span>")
				H.IgniteMob()
	else
		if(H.fire_stacks)
			var/obj/item/clothing/suit/space/eva/plasmaman/P = H.wear_suit
			if(istype(P))
				P.Extinguish(H)
	H.update_fire()
```